File: app/src/main/java/calendario/kevshupp/diariokevinali/DateFilterAdapter.java
package calendario.kevshupp.diariokevinali;

import android.graphics.Color;
import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.core.content.res.ResourcesCompat;
import androidx.recyclerview.widget.RecyclerView;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class DateFilterAdapter extends RecyclerView.Adapter<DateFilterAdapter.DateViewHolder> {

    private List<Long> dateList;
    private OnDateSelectedListener listener;
    private SimpleDateFormat sdf = new SimpleDateFormat("EEEE, dd 'de' MMMM", new Locale("es", "ES"));
    private SimpleDateFormat yearSdf = new SimpleDateFormat("yyyy", Locale.getDefault());

    public interface OnDateSelectedListener {
        void onDateSelected(long timestamp);
    }

    public DateFilterAdapter(List<Long> dateList, OnDateSelectedListener listener) {
        this.dateList = dateList;
        this.listener = listener;
    }

    @NonNull
    @Override
    public DateViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_date_filter, parent, false);
        return new DateViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DateViewHolder holder, int position) {
        long timestamp = dateList.get(position);
        Date date = new Date(timestamp);
        
        String formattedDate = sdf.format(date);
        // Capitalize first letter
        formattedDate = formattedDate.substring(0, 1).toUpperCase() + formattedDate.substring(1);
        
        holder.tvDate.setText(formattedDate);
        holder.tvYear.setText(yearSdf.format(date));
        
        holder.itemView.setOnClickListener(v -> listener.onDateSelected(timestamp));
    }

    @Override
    public int getItemCount() {
        return dateList.size();
    }

    static class DateViewHolder extends RecyclerView.ViewHolder {
        TextView tvDate, tvYear;
        public DateViewHolder(@NonNull View itemView) {
            super(itemView);
            tvDate = itemView.findViewById(R.id.tvFilterDateText);
            tvYear = itemView.findViewById(R.id.tvFilterYearText);
            
            Typeface pixelFont;
            try {
                pixelFont = ResourcesCompat.getFont(itemView.getContext(), R.font.vt323);
                tvDate.setTypeface(pixelFont);
                tvYear.setTypeface(pixelFont);
            } catch (Exception ignored) {}
        }
    }
}

File: app/src/main/java/calendario/kevshupp/diariokevinali/DateFilterAdapter.kt
package calendario.kevshupp.diariokevinali

import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.res.ResourcesCompat
import androidx.recyclerview.widget.RecyclerView
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class DateFilterAdapter(
    private val dateList: List<Long>,
    private val listener: OnDateSelectedListener
) : RecyclerView.Adapter<DateFilterAdapter.DateViewHolder>() {

    private val sdf = SimpleDateFormat("EEEE, dd 'de' MMMM", Locale("es", "ES"))
    private val yearSdf = SimpleDateFormat("yyyy", Locale.getDefault())

    interface OnDateSelectedListener {
        fun onDateSelected(timestamp: Long)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DateViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_date_filter, parent, false)
        return DateViewHolder(view)
    }

    override fun onBindViewHolder(holder: DateViewHolder, position: Int) {
        val timestamp = dateList[position]
        val date = Date(timestamp)

        var formattedDate = sdf.format(date)
        if (formattedDate.isNotEmpty()) {
            formattedDate = formattedDate.substring(0, 1).uppercase() + formattedDate.substring(1)
        }

        holder.tvDate.text = formattedDate
        holder.tvYear.text = yearSdf.format(date)

        holder.itemView.setOnClickListener { listener.onDateSelected(timestamp) }
    }

    override fun getItemCount(): Int = dateList.size

    class DateViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvDate: TextView = itemView.findViewById(R.id.tvFilterDateText)
        val tvYear: TextView = itemView.findViewById(R.id.tvFilterYearText)

        init {
            try {
                val pixelFont: Typeface? = ResourcesCompat.getFont(itemView.context, R.font.vt323)
                tvDate.typeface = pixelFont
                tvYear.typeface = pixelFont
            } catch (ignored: Exception) {
            }
        }
    }
}
